File: packages/core/src/routes/review.ts
/**
 * Review queue route — pending bullets and perspectives.
 */

import { Hono } from 'hono'
import type { Services } from '../services'
import { mapStatusCode } from './server'

export function reviewRoutes(services: Services) {
  const app = new Hono()

  app.get('/review/pending', async (c) => {
    const result = await services.review.getPendingReview()
    if (!result.ok) return c.json({ error: result.error }, mapStatusCode(result.error.code))
    return c.json({ data: result.data })
  })

  return app
}

File: crates/forge-server/src/routes/review.rs
//! Review queue route — pending bullets and perspectives.
//!
//! Mirrors `packages/core/src/routes/review.ts`.

use axum::extract::State;
use axum::routing::get;
use axum::{Json, Router};

use forge_core::ReviewQueue;
use forge_sdk::services::ReviewService;

use crate::db::with_conn;
use crate::error::ApiError;
use crate::response::ApiData;
use crate::state::SharedState;

// ── Handlers ────────────────────────────────────────────────────────

async fn get_pending_review(
    State(state): State<SharedState>,
) -> Result<Json<ApiData<ReviewQueue>>, ApiError> {
    let result = with_conn(&state, move |conn| {
        ReviewService::get_pending_review(conn)
    })
    .await?;
    Ok(Json(ApiData { data: result }))
}

// ── Router ──────────────────────────────────────────────────────────

pub fn router() -> Router<SharedState> {
    Router::new()
        .route("/review/pending", get(get_pending_review))
}
